### ISSUE TITLE
500 Internal Server Error when deleting Course Section

### ISSUE THREAD

**Issue by rickhumphries** [elmsln/elmsln#2497]
Bug:
- v0.8.1

I'm getting an Internal Server error when trying to remove a course section from a course in the CIS. The error message coming through on NewRelic is:

```
PDOException
/{coursename}/elmsln.php

Error message
PDOException: Uncaught exception 'PDOException' with message 'SQLSTATE[HY000]: General error: 1267 Illegal mix of collations (ascii_bin,IMPLICIT) and (utf8_general_ci,COERCIBLE) for operation '='' in /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/database/database.inc:2227

Stack trace
in PDOStatement::execute called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/database/database.inc (2227)
in DatabaseStatementBase::execute called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/database/database.inc (697)
in DatabaseConnection::query called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/database/query.inc (858)
in DeleteQuery::execute called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/cache.inc (552)
in DrupalDatabaseCache::clear called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/apdqc/apdqc.cache.inc (624)
in APDQCache::clear called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/cache.inc (173)
in cache_clear_all called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathcache/pathcache.module (28)
in pathcache_clear_cache called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathcache/pathcache.module (36)
in pathcache_path_insert called at ? (?)
in call_user_func_array called at /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/module.inc (957)
in module_invoke_all called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathcache/path.inc (527)
in path_save called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathauto/pathauto.inc (563)
in _pathauto_set_alias called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathauto/pathauto.inc (441)
in pathauto_create_alias called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathauto/pathauto.module (1091)
in pathauto_user_update_alias called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/ulmus/pathauto/pathauto.module (1030)
in pathauto_user_insert called at /var/www/elmsln/core/dslmcode/cores/drupal-7/modules/user/user.module (101)
in user_module_invoke called at /var/www/elmsln/core/dslmcode/cores/drupal-7/modules/user/user.module (592)
…b/cis_connector/modules/features/cis_section/src/CisSectionRosterProcessor.php (261)
…b/cis_connector/modules/features/cis_section/src/CisSectionRosterProcessor.php (191)
…b/cis_connector/modules/features/cis_section/src/CisSectionRosterProcessor.php (136)
in CisSectionRosterProcessor::process called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/cis_section/cis_section.module (538)
in _cis_section_create_accounts called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/services/cis_course_service/callbacks/v1/sync_data_model.inc (37)
in cis_course_service_elmsln_api_callback_sync_data_model called at ? (?)
in call_user_func_array called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/core/elmsln_api/includes/callback.inc (32)
in elmsln_api_callback_execute called at /var/www/elmsln/core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/core/elmsln_api/elmsln_api.module (1033)
…cute_request called at /var/www/elmsln/core/dslmcode/cores/drupal-7/elmsln.php (27)
```

**Comment by btopro** [comment]
going to guess this is still a thing since mine just spins forever.

**Comment by btopro** [comment]
it may sound crazy, but i dono if we've ever deleted a section in the several years using this cause ours are created via script largely.

**Comment by btopro** [comment]
said error does NOT happen if you delete from the course page by removing the whole offering (which will then get rid of the section). I don't know if you use multiple offerings or not but if there's only 1 section in an offering then you can get rid of it from that page (it only lists the month/year of the term)

**Comment by rickhumphries** [comment]
Ah dammit. In this case I've got 8 sections linked to one offering so I don't think I can delete the whole thing.

**Comment by btopro** [comment]
![2018-11-15_13-28-53](https://user-images.githubusercontent.com/329735/48574200-89f84b00-e8dc-11e8-8284-85421ea0356b.png)

**Comment by btopro** [comment]
duplicate of #2313 which was resolved in 0.10.0
